Content/GUI/SkillsTree/SkillsTreeInnerPanel.cs

```diff
@@ -1,27 +1,37 @@
 ﻿using PathOfTerraria.Core.Loaders.UILoading;
-using Terraria.UI;
+using PathOfTerraria.Core.Mechanics;
 
 namespace PathOfTerraria.Content.GUI.SkillsTree;
 
 internal class SkillsTreeInnerPanel : SmartUIElement
 {
-	private UIElement Panel => Parent;
 	public override string TabName => "SkillTree";
 
+	private bool _drewSkills;
+	
 	public override void Draw(SpriteBatch spriteBatch)
 	{
-		Utils.DrawBorderStringBig(
-			spriteBatch, 
-			"Skills - Placeholder",
-			GetRectangle().TopLeft() + new Vector2(138, 150),
-			Color.White,
-			0.6f,
-			0.5f,
-			0.35f);
+		base.Draw(spriteBatch);
+		if (!_drewSkills)
+		{
+			_drewSkills = true;
+			DrawAllSkills();
```

Comment:
For clarity's sake, I'd recommend calling methods like this that *append* children - *not* draw them - something like "AppendAllSkills" instead. This would read to me as a method that manually draws children every frame, not a one-time set for appending children.